### PR TITLE
Delete .NET Native funceval support

### PR DESCRIPTION
**Pull request by MichalStrehovsky**
* We don't have a debugger to take advantage of this right now.
* Depends on System.Private.Debug that was not open sourced.

### CODE REVIEW

File: src/coreclr/nativeaot/Runtime/threadstore.cpp [hunk 1]
```diff
@@ -325,7 +303,7 @@ void ThreadStore::WaitForSuspendComplete()
 
 void ThreadStore::InitiateThreadAbort(Thread* targetThread, Object * threadAbortException, bool doRudeAbort)
 {
-    SuspendAllThreads(/* waitForGCEvent = */ false, /* fireDebugEvent = */ false);
+    SuspendAllThreads(/* waitForGCEvent = */ false);
     // TODO: consider enabling multiple thread aborts running in parallel on different threads
```
Comment (jkotas) [inline src/coreclr/nativeaot/Runtime/threadstore.cpp:307]: Everything related to threadabort is also only for FuncEval.

**Review by jkotas — APPROVED** [review]
LGTM. Thanks

File: src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs [hunk 2]
```diff
@@ -1080,12 +1080,6 @@ public static void RhpSendCustomEventToDebugger(IntPtr payload, int length)
             RuntimeImports.RhpSendCustomEventToDebugger(payload, length);
```
Comment (jkotas) [inline src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/Augments/RuntimeAugments.cs:1080]: RhpSendCustomEventToDebugger is also only for FuncEval

File: src/coreclr/nativeaot/Runtime/threadstore.cpp (hunk 1 above)
Comment (MichalStrehovsky) [reply src/coreclr/nativeaot/Runtime/threadstore.cpp:307]: Looks like that would require undoing https://github.com/dotnet/corert/pull/4100/commits/a61fac9dce2b1873a61cbe70489d8c0ecdad3ad4. It's a bit convoluted, so maybe a separate pull request at another time.

**Pull request merged**